test/nodes/test_observability.py

```
@@ -0,0 +1,422 @@
+        REGISTRY = prometheus_metrics.REGISTRY
+        before = REGISTRY.get_sample_value('rocketride_llm_requests_total', {'provider': 'openai', 'model': 'gpt-4', 'status': 'success'}) or 0.0
+        prometheus_metrics.LLM_REQUESTS.labels(provider='openai', model='gpt-4', status='success').inc()
+        after = REGISTRY.get_sample_value('rocketride_llm_requests_total', {'provider': 'openai', 'model': 'gpt-4', 'status': 'success'})
+        assert after == before + 1.0
+
+    def test_llm_tokens_counter_by_type(self):
+        REGISTRY = prometheus_metrics.REGISTRY
+        before_in = REGISTRY.get_sample_value('rocketride_llm_tokens_total', {'provider': 'openai', 'model': 'gpt-4', 'token_type': 'input'}) or 0.0
+        before_out = REGISTRY.get_sample_value('rocketride_llm_tokens_total', {'provider': 'openai', 'model': 'gpt-4', 'token_type': 'output'}) or 0.0
+
+        prometheus_metrics.LLM_TOKENS.labels(provider='openai', model='gpt-4', token_type='input').inc(100)
```

Ruff (0.15.7)

[error] Possible hardcoded password assigned to argument: "token_type"

(S106)